Qt ts规则 触发 windows命令行字符串限制

### Xmake 版本

3.0.1

### 操作系统版本和架构

win11 22H2

### 描述问题

项目源文件越来越多，lupdate的关于源文件的参数越长，现在触发了windows命令行字符串限制

![Image](https://github.com/user-attachments/assets/285a5ca4-5b5a-4cd0-a8ef-12c3e3bca9db)

### 期待的结果

正常更新ts

可用解决方案：
lupdate.exe  -project .\file.json

json 内容：

[file.json](https://github.com/user-attachments/files/20863460/file.json)

### 工程配置

```lua
```

### 附加信息和错误日志

生成开始于 18:39...
1>------ 已启动生成: 项目: GlueHmi, 配置: minsizerel x64 ------
1>$xmake config -y  -P . -p windows -m minsizerel -a x64 -o "build" --qt=D:/Qt/5.15.2/msvc2019_64 --qt_sdkver=5.15.2
1>checking for Microsoft Visual Studio (x64) version ... 2022
1>checking for Microsoft C/C++ Compiler (x64) version ... 19.44.35211
1>checking for Qt SDK directory ... D:/Qt/5.15.2/msvc2019_64
1>checking for Qt SDK version ... 5.15.2
1>EXEC : error : cannot runv(D:\Qt\5.15.2\msvc2019_64\bin\lupdate.exe -no-obsolete src\resource\IconTheme\style_icon-Zorin.qrc src\resource\IconTheme\style_icon-ZorinBlue-Dark.qrc src\resource\IconTheme\style_icon-ZorinBlue-Light.qrc src\app\Resource\Icons.qrc src\app\Resource\Resource.qrc src\app\Resource\UIResource.qrc src\app\Core\AppInfo\AppInfo.cpp src\app\Core\FileDialog\FileDialog.cpp src\app\Core\FileDialog\IPreviewWidget.cpp src\app\Core\FileDialog\PublicFileDialog.cpp src\app\Core\Language\LanguageHelper.cpp src\app\Core\Language\Translator.cpp src\app\Core\Logger\Logger.cpp src\app\Core\SettingsManager\SettingsManager.cpp src\app\Core\SingleApplication\SingleApplication.cpp src\app\Core\Theme\ModernPlatformStyle.cpp src\app\Core\Theme\StyledItemDelegate.cpp src\app\Core\Theme\ThemeManager.cpp src\app\Core\UserManager\UserManager.cpp src\app\Core\UserManager\Widgets\UserLogInDlg.cpp src\app\Core\UserManager\Widgets\UserManagerDlg.cpp src\app\Core\UserManager\Widgets\UserPWDResetDlg.cpp src\app\main.cpp src\app\Module\Alarms\AlarmsDialog.cpp src\app\Module\Alarms\AlarmsManager.cpp src\app\Module\Base\Dialog.cpp src\app\Module\Base\MsgBox.cpp src\app\Module\Base\VariantComboBox.cpp src\app\Module\Clipper2Ex\Clipper2Ex.cpp src\app\Module\Clipper2Ex\Polylabel.cpp src\app\Module\Clipper2Ex\PolyOffseter.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXY.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXYEx.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionControlTable.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionTable.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentButton.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentDialog.cpp src\app\Module\CommonWidgets\AxisControl\FineTuningStep.cpp src\app\Module\CommonWidgets\AxisControl\ItemDelegate.cpp src\app\Module\CommonWidgets\Ruler\mRuler.cpp src\app\Module\CommonWidgets\Ruler\RulerWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\GaugePanel.cpp src\app\Module\CommonWidgets\SpeedMeter\ObsSpeedMeterWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\SpeedMeterWidget.cpp src\app\Module\CommonWidgets\ViewWidget\TableWidget.cpp src\app\Module\EmbCanvas\ElidedLabel.cpp src\app\Module\EmbCanvas\EmbCanvasController.cpp src\app\Module\EmbCanvas\EmbCanvasMove.cpp src\app\Module\EmbCanvas\EmbCanvasPreviewWidget.cpp src\app\Module\EmbCanvas\EmbCanvasWidget.cpp src\app\Module\EmbCanvas\EmbImportStrategy.cpp src\app\Module\EmbCanvas\GlueCodeComboBox.cpp src\app\Module\Factory\FactoryRegisterHelper.cpp src\app\Module\Graphics\GraphicsCrossItem.cpp src\app\Module\Graphics\GraphicsMeasureItem.cpp src\app\Module\Graphics\GraphicsPointsItem.cpp src\app\Module\Graphics\GraphicsRectResizeHandler.cpp src\app\Module\Graphics\GraphicsScene.cpp src\app\Module\Graphics\GraphicsSelectGridItem.cpp src\app\Module\Graphics\GraphicsSelectRectItem.cpp src\app\Module\Graphics\GraphicsView.cpp src\app\Module\Graphics\OpenGL\GLWidget.cpp src\app\Module\Graphics\OpenGL\ImageGLWidget.cpp src\app\Module\Graphics\OpenGL\Renderer.cpp src\app\Module\Graphics\QGraphicsImageItem.cpp src\app\Module\Graphics\SceneTreeItem.cpp src\app\Module\Graphics\SceneTreeView.cpp src\app\Module\MindVision\CameraGraphicsScene.cpp src\app\Module\MindVision\CameraManager.cpp src\app\Module\MindVision\MindVisionCamera.cpp src\app\Module\MindVision\MindVisionCameraInfo.cpp src\app\Module\MindVision\VideoUtils.cpp src\app\Module\Modbus\DriversAddrListManager.cpp src\app\Module\Modbus\ModbusHelper.cpp src\app\Module\Modbus\SerialPortHelper.cpp src\app\Module\ModuleManager\FunctionCheckBox.cpp src\app\Module\ModuleManager\IFunctionWidget.cpp src\app\Module\ModuleManager\ModuleHelper.cpp src\app\Module\ModuleManager\ModuleManager.cpp src\app\Module\ModuleManager\ObsFunctionCheckBox.cpp src\app\Module\ModuleManager\ObsFunctionRadioButton.cpp src\app\Module\MountWatcher\MountWatcher.cpp src\app\Module\Register\RegInfoOld.cpp src\app\Module\Register\RegisterManager.cpp src\app\Module\Register\SoftRegNew.cpp src\app\Module\Register\SoftRegOld.cpp src\app\Module\Register\Widgets\RegisterWidget.cpp src\app\Module\SoundSetting\SoundService.cpp src\app\Module\Units\ColorHelper.cpp src\app\Module\Units\FileHelper.cpp src\app\Module\Units\GraphicHelper.cpp src\app\Module\Units\ImageHelper.cpp src\app\Module\Units\PathHelper.cpp src\app\Module\Units\ScreenShot.cpp src\app\Module\Units\SqlHelper.cpp src\app\Module\Units\ZipHelper.cpp src\app\Module\Upgrade\TSUpgradeHelper.cpp src\app\Module\VisionSystem\BaseSystem.cpp src\app\Module\VisionSystem\CalibrationSystem.cpp src\app\Module\VisionSystem\Component\ImageProcessor.cpp src\app\Module\VisionSystem\Component\InputComponent.cpp src\app\Module\VisionSystem\Component\LogComponent.cpp src\app\Module\VisionSystem\Component\ObjectLocatorComponent.cpp src\app\Module\VisionSystem\Component\ProcessComponent.cpp src\app\Module\VisionSystem\Component\ShapeArrayToolComponent.cpp src\app\Module\VisionSystem\Component\WorkspaceInput.cpp src\app\Module\VisionSystem\Data\PrimitiveConvertor2.cpp src\app\Module\VisionSystem\Data\ShapeConfig.cpp src\app\Module\VisionSystem\Data\VisionConfig.cpp src\app\Module\VisionSystem\Data\VisionMeta.cpp src\app\Module\VisionSystem\Data\VisionMetaFileCodec.cpp src\app\Module\VisionSystem\Graphics\DoubleGraphicsView.cpp src\app\Module\VisionSystem\Graphics\EdgeItemDelegate.cpp src\app\Module\VisionSystem\Graphics\GraphicsToolBar.cpp src\app\Module\VisionSystem\Graphics\ShapeTableModel.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsScene.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsView.cpp src\app\Module\VisionSystem\Graphics\VisionSystemGraphics.cpp src\app\Module\VisionSystem\Manager\CalibManager.cpp src\app\Module\VisionSystem\Manager\EdgesTools.cpp src\app\Module\VisionSystem\Manager\ShapeManager.cpp src\app\Module\VisionSystem\Manager\VisionSystemManager.cpp src\app\Module\VisionSystem\Manager\WorkSpaceManager.cpp src\app\Module\VisionSystem\OpenCV\CvCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvCalibTool.cpp src\app\Module\VisionSystem\OpenCV\CvEdgePatrol.cpp src\app\Module\VisionSystem\OpenCV\CvFixedCameraCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvObjectLocator.cpp src\app\Module\VisionSystem\OpenCV\CvShapeArrayTool.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTools.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTypes.cpp src\app\Module\VisionSystem\Process\EdgeProcess.cpp src\app\Module\VisionSystem\Process\WarpProcess.cpp src\app\Module\VisionSystem\Units\AngleDistanceTree.cpp src\app\Module\VisionSystem\Units\VisionGraphicsTools.cpp src\app\Module\VisionSystem\VisionAutoWork.cpp src\app\Module\VisionSystem\VisionSystem.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibration.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibTool.cpp src\app\Module\VisionSystem\ZSVision\ZsEdgePatrol.cpp src\app\Module\VisionSystem\ZSVision\ZsObjectLocator.cpp src\app\Module\VisionSystem\ZSVision\ZsShapeArrayTool.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionManager.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionTypes.cpp src\app\Module\WorkData\WorkDataHelper.cpp src\app\Module\XModem\FileDevice.cpp src\app\Module\XModem\Transport.cpp src\app\Module\XModem\XModem.cpp src\app\Module\XModem\XModemHelper.cpp src\app\UI\DebugPage\AxisMotion\AxisMotion.cpp src\app\UI\DebugPage\Drivers\DriversPages.cpp src\app\UI\DebugPage\Drivers\PulsesObsLabel.cpp src\app\UI\DebugPage\IOControl\IOController.cpp src\app\UI\DebugPage\IOControl\ObsControllerButton.cpp src\app\UI\DebugPage\IOControl\ObsIOPushButton.cpp src\app\UI\DebugPage\MaterialTransfer\MaterialTransferPage.cpp src\app\UI\DebugPage\RunningInfo\DataSelectDialog.cpp src\app\UI\DebugPage\RunningInfo\RunningInfoPage.cpp src\app\UI\DebugPage\Workstation\WorkstationPage.cpp src\app\UI\EditorPage\CodeColorDlg.cpp src\app\UI\EditorPage\CodeInputDlg.cpp src\app\UI\EditorPage\CoordinatePointInputDlg.cpp src\app\UI\EditorPage\CopyDlg.cpp src\app\UI\EditorPage\EmbCanvasEditorPage.cpp src\app\UI\EditorPage\layers\centeredcheckboxdelegate.cpp src\app\UI\EditorPage\layers\changelayerdialog.cpp src\app\UI\EditorPage\layers\layerdata.cpp src\app\UI\EditorPage\layers\layermodel.cpp src\app\UI\EditorPage\layers\layerpage.cpp src\app\UI\EditorPage\layers\layerservice.cpp src\app\UI\EditorPage\LayerSetDlg.cpp src\app\UI\EditorPage\ParameterDlg.cpp src\app\UI\GeneralPage\Location\BasicPositionPage.cpp src\app\UI\GeneralPage\Location\NeedleDialog.cpp src\app\UI\GeneralPage\Process\FileList\FileListHelper.cpp src\app\UI\GeneralPage\Process\FileList\FileListWidget.cpp src\app\UI\GeneralPage\Process\MainPage.cpp src\app\UI\MainWindow\MainWindow.cpp src\app\UI\MainWindow\State\StateWidget.cpp src\app\UI\MainWindow\Title\TitleWidget.cpp src\app\UI\ManagePage\Other\OtherPage.cpp src\app\UI\ManagePage\Software\ExportSelection\ExportSelectionDlg.cpp src\app\UI\ManagePage\Software\SoftwarePage.cpp src\app\UI\ManagePage\WorkRecord\WorkRecordPage.cpp src\app\UI\Shared\ProcessingControlWidget.cpp src\app\UI\Shared\RunInfo\RunInfoWidget.cpp src\app\UI\Shared\SharedControl.cpp src\app\UI\Shared\WorkAssist\WorkAssistWidget.cpp src\app\UI\SplashScreen\SplashScreen.cpp src\app\UI\TabWidgetPage\ModuleTabWidget.cpp src\app\UI\TabWidgetPage\Navigation\LockScreen\LockScreenDlg.cpp src\app\UI\TabWidgetPage\Navigation\NavigationWidget.cpp src\app\UI\TabWidgetPage\TabWidgetPages.cpp src\app\UI\VisionPage\Calibration\CalibrationPage.cpp src\app\UI\VisionPage\Camera\CameraDoublePage.cpp src\app\UI\VisionPage\Camera\CameraPage.cpp src\app\UI\VisionPage\Camera\CameraWidget.cpp src\app\UI\VisionPage\Share\RegionToolBar.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\EdgePatrolConfigDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\ImportShapeDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\EdgePatrolConfigWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\ShapeManagerWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\VisionDataEditWidget.cpp src\app\UI\VisionPage\Vision\Log\VisionLogWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainControlWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainPage.cpp src\app\UI\VisionPage\Vision\Main\VisionPage.cpp src\app\UI\VisionPage\Vision\Options\VisionAdvancedDlg.cpp src\app\UI\VisionPage\Vision\Options\VisionOptionsDlg.cpp src\app\Core\AppInfo\AppInfo.cpp src\app\Core\FileDialog\FileDialog.cpp src\app\Core\FileDialog\IPreviewWidget.cpp src\app\Core\FileDialog\PublicFileDialog.cpp src\app\Core\Language\LanguageHelper.cpp src\app\Core\Language\Translator.cpp src\app\Core\Logger\Logger.cpp src\app\Core\SettingsManager\SettingsManager.cpp src\app\Core\SingleApplication\SingleApplication.cpp src\app\Core\Theme\ModernPlatformStyle.cpp src\app\Core\Theme\StyledItemDelegate.cpp src\app\Core\Theme\ThemeManager.cpp src\app\Core\UserManager\UserManager.cpp src\app\Core\UserManager\Widgets\UserLogInDlg.cpp src\app\Core\UserManager\Widgets\UserManagerDlg.cpp src\app\Core\UserManager\Widgets\UserPWDResetDlg.cpp src\app\main.cpp src\app\Module\Alarms\AlarmsDialog.cpp src\app\Module\Alarms\AlarmsManager.cpp src\app\Module\Base\Dialog.cpp src\app\Module\Base\MsgBox.cpp src\app\Module\Base\VariantComboBox.cpp src\app\Module\Clipper2Ex\Clipper2Ex.cpp src\app\Module\Clipper2Ex\Polylabel.cpp src\app\Module\Clipper2Ex\PolyOffseter.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXY.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXYEx.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionControlTable.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionTable.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentButton.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentDialog.cpp src\app\Module\CommonWidgets\AxisControl\FineTuningStep.cpp src\app\Module\CommonWidgets\AxisControl\ItemDelegate.cpp src\app\Module\CommonWidgets\Ruler\mRuler.cpp src\app\Module\CommonWidgets\Ruler\RulerWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\GaugePanel.cpp src\app\Module\CommonWidgets\SpeedMeter\ObsSpeedMeterWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\SpeedMeterWidget.cpp src\app\Module\CommonWidgets\ViewWidget\TableWidget.cpp src\app\Module\EmbCanvas\ElidedLabel.cpp src\app\Module\EmbCanvas\EmbCanvasController.cpp src\app\Module\EmbCanvas\EmbCanvasMove.cpp src\app\Module\EmbCanvas\EmbCanvasPreviewWidget.cpp src\app\Module\EmbCanvas\EmbCanvasWidget.cpp src\app\Module\EmbCanvas\EmbImportStrategy.cpp src\app\Module\EmbCanvas\GlueCodeComboBox.cpp src\app\Module\Factory\FactoryRegisterHelper.cpp src\app\Module\Graphics\GraphicsCrossItem.cpp src\app\Module\Graphics\GraphicsMeasureItem.cpp src\app\Module\Graphics\GraphicsPointsItem.cpp src\app\Module\Graphics\GraphicsRectResizeHandler.cpp src\app\Module\Graphics\GraphicsScene.cpp src\app\Module\Graphics\GraphicsSelectGridItem.cpp src\app\Module\Graphics\GraphicsSelectRectItem.cpp src\app\Module\Graphics\GraphicsView.cpp src\app\Module\Graphics\OpenGL\GLWidget.cpp src\app\Module\Graphics\OpenGL\ImageGLWidget.cpp src\app\Module\Graphics\OpenGL\Renderer.cpp src\app\Module\Graphics\QGraphicsImageItem.cpp src\app\Module\Graphics\SceneTreeItem.cpp src\app\Module\Graphics\SceneTreeView.cpp src\app\Module\MindVision\CameraGraphicsScene.cpp src\app\Module\MindVision\CameraManager.cpp src\app\Module\MindVision\MindVisionCamera.cpp src\app\Module\MindVision\MindVisionCameraInfo.cpp src\app\Module\MindVision\VideoUtils.cpp src\app\Module\Modbus\DriversAddrListManager.cpp src\app\Module\Modbus\ModbusHelper.cpp src\app\Module\Modbus\SerialPortHelper.cpp src\app\Module\ModuleManager\FunctionCheckBox.cpp src\app\Module\ModuleManager\IFunctionWidget.cpp src\app\Module\ModuleManager\ModuleHelper.cpp src\app\Module\ModuleManager\ModuleManager.cpp src\app\Module\ModuleManager\ObsFunctionCheckBox.cpp src\app\Module\ModuleManager\ObsFunctionRadioButton.cpp src\app\Module\MountWatcher\MountWatcher.cpp src\app\Module\Register\RegInfoOld.cpp src\app\Module\Register\RegisterManager.cpp src\app\Module\Register\SoftRegNew.cpp src\app\Module\Register\SoftRegOld.cpp src\app\Module\Register\Widgets\RegisterWidget.cpp src\app\Module\SoundSetting\SoundService.cpp src\app\Module\Units\ColorHelper.cpp src\app\Module\Units\FileHelper.cpp src\app\Module\Units\GraphicHelper.cpp src\app\Module\Units\ImageHelper.cpp src\app\Module\Units\PathHelper.cpp src\app\Module\Units\ScreenShot.cpp src\app\Module\Units\SqlHelper.cpp src\app\Module\Units\ZipHelper.cpp src\app\Module\Upgrade\TSUpgradeHelper.cpp src\app\Module\VisionSystem\BaseSystem.cpp src\app\Module\VisionSystem\CalibrationSystem.cpp src\app\Module\VisionSystem\Component\ImageProcessor.cpp src\app\Module\VisionSystem\Component\InputComponent.cpp src\app\Module\VisionSystem\Component\LogComponent.cpp src\app\Module\VisionSystem\Component\ObjectLocatorComponent.cpp src\app\Module\VisionSystem\Component\ProcessComponent.cpp src\app\Module\VisionSystem\Component\ShapeArrayToolComponent.cpp src\app\Module\VisionSystem\Component\WorkspaceInput.cpp src\app\Module\VisionSystem\Data\PrimitiveConvertor2.cpp src\app\Module\VisionSystem\Data\ShapeConfig.cpp src\app\Module\VisionSystem\Data\VisionConfig.cpp src\app\Module\VisionSystem\Data\VisionMeta.cpp src\app\Module\VisionSystem\Data\VisionMetaFileCodec.cpp src\app\Module\VisionSystem\Graphics\DoubleGraphicsView.cpp src\app\Module\VisionSystem\Graphics\EdgeItemDelegate.cpp src\app\Module\VisionSystem\Graphics\GraphicsToolBar.cpp src\app\Module\VisionSystem\Graphics\ShapeTableModel.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsScene.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsView.cpp src\app\Module\VisionSystem\Graphics\VisionSystemGraphics.cpp src\app\Module\VisionSystem\Manager\CalibManager.cpp src\app\Module\VisionSystem\Manager\EdgesTools.cpp src\app\Module\VisionSystem\Manager\ShapeManager.cpp src\app\Module\VisionSystem\Manager\VisionSystemManager.cpp src\app\Module\VisionSystem\Manager\WorkSpaceManager.cpp src\app\Module\VisionSystem\OpenCV\CvCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvCalibTool.cpp src\app\Module\VisionSystem\OpenCV\CvEdgePatrol.cpp src\app\Module\VisionSystem\OpenCV\CvFixedCameraCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvObjectLocator.cpp src\app\Module\VisionSystem\OpenCV\CvShapeArrayTool.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTools.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTypes.cpp src\app\Module\VisionSystem\Process\EdgeProcess.cpp src\app\Module\VisionSystem\Process\WarpProcess.cpp src\app\Module\VisionSystem\Units\AngleDistanceTree.cpp src\app\Module\VisionSystem\Units\VisionGraphicsTools.cpp src\app\Module\VisionSystem\VisionAutoWork.cpp src\app\Module\VisionSystem\VisionSystem.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibration.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibTool.cpp src\app\Module\VisionSystem\ZSVision\ZsEdgePatrol.cpp src\app\Module\VisionSystem\ZSVision\ZsObjectLocator.cpp src\app\Module\VisionSystem\ZSVision\ZsShapeArrayTool.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionManager.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionTypes.cpp src\app\Module\WorkData\WorkDataHelper.cpp src\app\Module\XModem\FileDevice.cpp src\app\Module\XModem\Transport.cpp src\app\Module\XModem\XModem.cpp src\app\Module\XModem\XModemHelper.cpp src\app\UI\DebugPage\AxisMotion\AxisMotion.cpp src\app\UI\DebugPage\Drivers\DriversPages.cpp src\app\UI\DebugPage\Drivers\PulsesObsLabel.cpp src\app\UI\DebugPage\IOControl\IOController.cpp src\app\UI\DebugPage\IOControl\ObsControllerButton.cpp src\app\UI\DebugPage\IOControl\ObsIOPushButton.cpp src\app\UI\DebugPage\MaterialTransfer\MaterialTransferPage.cpp src\app\UI\DebugPage\RunningInfo\DataSelectDialog.cpp src\app\UI\DebugPage\RunningInfo\RunningInfoPage.cpp src\app\UI\DebugPage\Workstation\WorkstationPage.cpp src\app\UI\EditorPage\CodeColorDlg.cpp src\app\UI\EditorPage\CodeInputDlg.cpp src\app\UI\EditorPage\CoordinatePointInputDlg.cpp src\app\UI\EditorPage\CopyDlg.cpp src\app\UI\EditorPage\EmbCanvasEditorPage.cpp src\app\UI\EditorPage\layers\centeredcheckboxdelegate.cpp src\app\UI\EditorPage\layers\changelayerdialog.cpp src\app\UI\EditorPage\layers\layerdata.cpp src\app\UI\EditorPage\layers\layermodel.cpp src\app\UI\EditorPage\layers\layerpage.cpp src\app\UI\EditorPage\layers\layerservice.cpp src\app\UI\EditorPage\LayerSetDlg.cpp src\app\UI\EditorPage\ParameterDlg.cpp src\app\UI\GeneralPage\Location\BasicPositionPage.cpp src\app\UI\GeneralPage\Location\NeedleDialog.cpp src\app\UI\GeneralPage\Process\FileList\FileListHelper.cpp src\app\UI\GeneralPage\Process\FileList\FileListWidget.cpp src\app\UI\GeneralPage\Process\MainPage.cpp src\app\UI\MainWindow\MainWindow.cpp src\app\UI\MainWindow\State\StateWidget.cpp src\app\UI\MainWindow\Title\TitleWidget.cpp src\app\UI\ManagePage\Other\OtherPage.cpp src\app\UI\ManagePage\Software\ExportSelection\ExportSelectionDlg.cpp src\app\UI\ManagePage\Software\SoftwarePage.cpp src\app\UI\ManagePage\WorkRecord\WorkRecordPage.cpp src\app\UI\Shared\ProcessingControlWidget.cpp src\app\UI\Shared\RunInfo\RunInfoWidget.cpp src\app\UI\Shared\SharedControl.cpp src\app\UI\Shared\WorkAssist\WorkAssistWidget.cpp src\app\UI\SplashScreen\SplashScreen.cpp src\app\UI\TabWidgetPage\ModuleTabWidget.cpp src\app\UI\TabWidgetPage\Navigation\LockScreen\LockScreenDlg.cpp src\app\UI\TabWidgetPage\Navigation\NavigationWidget.cpp src\app\UI\TabWidgetPage\TabWidgetPages.cpp src\app\UI\VisionPage\Calibration\CalibrationPage.cpp src\app\UI\VisionPage\Camera\CameraDoublePage.cpp src\app\UI\VisionPage\Camera\CameraPage.cpp src\app\UI\VisionPage\Camera\CameraWidget.cpp src\app\UI\VisionPage\Share\RegionToolBar.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\EdgePatrolConfigDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\ImportShapeDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\EdgePatrolConfigWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\ShapeManagerWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\VisionDataEditWidget.cpp src\app\UI\VisionPage\Vision\Log\VisionLogWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainControlWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainPage.cpp src\app\UI\VisionPage\Vision\Main\VisionPage.cpp src\app\UI\VisionPage\Vision\Options\VisionAdvancedDlg.cpp src\app\UI\VisionPage\Vision\Options\VisionOptionsDlg.cpp src\app\Core\AppInfo\AppInfo.h src\app\Core\FileDialog\FileDialog.h src\app\Core\FileDialog\IPreviewWidget.h src\app\Core\FileDialog\PublicFileDialog.h src\app\Core\Language\LanguageHelper.h src\app\Core\Language\Translator.h src\app\Core\Logger\initCrashPad.h src\app\Core\Logger\Logger.h src\app\Core\SettingsManager\SettingsManager.h src\app\Core\SingleApplication\SingleApplication.h src\app\Core\Theme\ModernPlatformStyle.h src\app\Core\Theme\StyledItemDelegate.h src\app\Core\Theme\ThemeManager.h src\app\Core\UserManager\UserManager.h src\app\Core\UserManager\Widgets\UserLogInDlg.h src\app\Core\UserManager\Widgets\UserManagerDlg.h src\app\Core\UserManager\Widgets\UserPWDResetDlg.h src\app\Module\Alarms\AlarmsDialog.h src\app\Module\Alarms\AlarmsManager.h src\app\Module\Base\Dialog.h src\app\Module\Base\MsgBox.h src\app\Module\Base\VariantComboBox.h src\app\Module\Clipper2Ex\Clipper2Ex.h src\app\Module\Clipper2Ex\Polylabel.h src\app\Module\Clipper2Ex\PolyOffseter.h src\app\Module\CommonWidgets\AxisControl\AxisControllerXY.h src\app\Module\CommonWidgets\AxisControl\AxisControllerXYEx.h src\app\Module\CommonWidgets\AxisControl\AxisMotionControlTable.h src\app\Module\CommonWidgets\AxisControl\AxisMotionTable.h src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentButton.h src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentDialog.h src\app\Module\CommonWidgets\AxisControl\FineTuningStep.h src\app\Module\CommonWidgets\AxisControl\ItemDelegate.h src\app\Module\CommonWidgets\Ruler\mRuler.h src\app\Module\CommonWidgets\Ruler\RulerWidget.h src\app\Module\CommonWidgets\SpeedMeter\GaugePanel.h src\app\Module\CommonWidgets\SpeedMeter\ObsSpeedMeterWidget.h src\app\Module\CommonWidgets\SpeedMeter\SpeedMeterWidget.h src\app\Module\CommonWidgets\ViewWidget\TableWidget.h src\app\Module\EmbCanvas\ElidedLabel.h src\app\Module\EmbCanvas\EmbCanvasController.h src\app\Module\EmbCanvas\EmbCanvasMove.h src\app\Module\EmbCanvas\EmbCanvasPreviewWidget.h src\app\Module\EmbCanvas\EmbCanvasWidget.h src\app\Module\EmbCanvas\EmbImportStrategy.h src\app\Module\EmbCanvas\GlueCodeComboBox.h src\app\Module\Factory\FactoryRegisterHelper.h src\app\Module\Graphics\GraphicsCrossItem.h src\app\Module\Graphics\GraphicsMeasureItem.h src\app\Module\Graphics\GraphicsPointsItem.h src\app\Module\Graphics\GraphicsRectResizeHandler.h src\app\Module\Graphics\GraphicsScene.h src\app\Module\Graphics\GraphicsSelectGridItem.h src\app\Module\Graphics\GraphicsSelectRectItem.h src\app\Module\Graphics\GraphicsType.h src\app\Module\Graphics\GraphicsView.h src\app\Module\Graphics\OpenGL\GLWidget.h src\app\Module\Graphics\OpenGL\ImageGLWidget.h src\app\Module\Graphics\OpenGL\Renderer.h src\app\Module\Graphics\QGraphicsImageItem.h src\app\Module\Graphics\SceneTreeItem.h src\app\Module\Graphics\SceneTreeView.h src\app\Module\MindVision\CameraGraphicsScene.h src\app\Module\MindVision\CameraManager.h src\app\Module\MindVision\MindVisionCamera.h src\app\Module\MindVision\MindVisionCameraInfo.h src\app\Module\MindVision\VideoUtils.h src\app\Module\Modbus\DriversAddrListManager.h src\app\Module\Modbus\ModbusHelper.h src\app\Module\Modbus\SerialPortHelper.h src\app\Module\ModuleManager\FunctionCheckBox.h src\app\Module\ModuleManager\IFunctionWidget.h src\app\Module\ModuleManager\ModuleHelper.h src\app\Module\ModuleManager\ModuleManager.h src\app\Module\ModuleManager\ObsFunctionCheckBox.h src\app\Module\ModuleManager\ObsFunctionRadioButton.h src\app\Module\MountWatcher\MountWatcher.h src\app\Module\Register\IRegister.h src\app\Module\Register\RegInfoOld.h src\app\Module\Register\RegisterManager.h src\app\Module\Register\SoftRegNew.h src\app\Module\Register\SoftRegOld.h src\app\Module\Register\Widgets\RegisterWidget.h src\app\Module\SoundSetting\SoundService.h src\app\Module\Units\ColorHelper.h src\app\Module\Units\DKTree.h src\app\Module\Units\FileHelper.h src\app\Module\Units\GraphicHelper.h src\app\Module\Units\ImageHelper.h src\app\Module\Units\PathHelper.h src\app\Module\Units\ScreenShot.h src\app\Module\Units\SqlHelper.h src\app\Module\Units\ZipHelper.h src\app\Module\Upgrade\TSUpgradeHelper.h src\app\Module\VisionSystem\BaseSystem.h src\app\Module\VisionSystem\CalibrationSystem.h src\app\Module\VisionSystem\Component\CalibComponent.h src\app\Module\VisionSystem\Component\CalibToolComponent.h src\app\Module\VisionSystem\Component\Component.h src\app\Module\VisionSystem\Component\EdgePatrolComponent.h src\app\Module\VisionSystem\Component\GraphicsComponent.h src\app\Module\VisionSystem\Component\ImageProcessor.h src\app\Module\VisionSystem\Component\InputComponent.h src\app\Module\VisionSystem\Component\LogComponent.h src\app\Module\VisionSystem\Component\ObjectLocatorComponent.h src\app\Module\VisionSystem\Component\ProcessComponent.h src\app\Module\VisionSystem\Component\ShapeArrayToolComponent.h src\app\Module\VisionSystem\Component\WorkspaceInput.h src\app\Module\VisionSystem\Data\PrimitiveConvertor2.h src\app\Module\VisionSystem\Data\ShapeConfig.h src\app\Module\VisionSystem\Data\VisionConfig.h src\app\Module\VisionSystem\Data\VisionMeta.h src\app\Module\VisionSystem\Data\VisionMetaFileCodec.h src\app\Module\VisionSystem\Graphics\DoubleGraphicsView.h src\app\Module\VisionSystem\Graphics\EdgeItemDelegate.h src\app\Module\VisionSystem\Graphics\GraphicsToolBar.h src\app\Module\VisionSystem\Graphics\ShapeTableModel.h src\app\Module\VisionSystem\Graphics\VisionGraphicsScene.h src\app\Module\VisionSystem\Graphics\VisionGraphicsView.h src\app\Module\VisionSystem\Graphics\VisionSystemGraphics.h src\app\Module\VisionSystem\Manager\CalibManager.h src\app\Module\VisionSystem\Manager\EdgesTools.h src\app\Module\VisionSystem\Manager\ShapeManager.h src\app\Module\VisionSystem\Manager\VisionSystemManager.h src\app\Module\VisionSystem\Manager\WorkSpaceManager.h src\app\Module\VisionSystem\OpenCV\CvCalibration.h src\app\Module\VisionSystem\OpenCV\CvCalibTool.h src\app\Module\VisionSystem\OpenCV\CvEdgePatrol.h src\app\Module\VisionSystem\OpenCV\CvFixedCameraCalibration.h src\app\Module\VisionSystem\OpenCV\CvObjectLocator.h src\app\Module\VisionSystem\OpenCV\CvShapeArrayTool.h src\app\Module\VisionSystem\OpenCV\OpenCvTools.h src\app\Module\VisionSystem\OpenCV\OpenCvTypes.h src\app\Module\VisionSystem\Process\EdgeProcess.h src\app\Module\VisionSystem\Process\WarpProcess.h src\app\Module\VisionSystem\Units\AngleDistanceTree.h src\app\Module\VisionSystem\Units\VisionGraphicsTools.h src\app\Module\VisionSystem\VisionAutoWork.h src\app\Module\VisionSystem\VisionSystem.h src\app\Module\VisionSystem\VisionSystemTypes.h src\app\Module\VisionSystem\ZSVision\ZsCalibration.h src\app\Module\VisionSystem\ZSVision\ZsCalibTool.h src\app\Module\VisionSystem\ZSVision\ZsEdgePatrol.h src\app\Module\VisionSystem\ZSVision\ZsObjectLocator.h src\app\Module\VisionSystem\ZSVision\ZsShapeArrayTool.h src\app\Module\VisionSystem\ZSVision\ZsVisionManager.h src\app\Module\VisionSystem\ZSVision\ZsVisionTypes.h src\app\Module\WorkData\WorkDataHelper.h src\app\Module\XModem\FileDevice.h src\app\Module\XModem\Transport.h src\app\Module\XModem\XModem.h src\app\Module\XModem\XModemHelper.h src\app\stable.h src\app\UI\DebugPage\AxisMotion\AxisMotion.h src\app\UI\DebugPage\Drivers\DriversPages.h src\app\UI\DebugPage\Drivers\PulsesObsLabel.h src\app\UI\DebugPage\IOControl\IOController.h src\app\UI\DebugPage\IOControl\ObsControllerButton.h src\app\UI\DebugPage\IOControl\ObsIOPushButton.h src\app\UI\DebugPage\MaterialTransfer\MaterialTransferPage.h src\app\UI\DebugPage\RunningInfo\DataSelectDialog.h src\app\UI\DebugPage\RunningInfo\RunningInfoPage.h src\app\UI\DebugPage\Workstation\WorkstationPage.h src\app\UI\EditorPage\CodeColorDlg.h src\app\UI\EditorPage\CodeInputDlg.h src\app\UI\EditorPage\CoordinatePointInputDlg.h src\app\UI\EditorPage\CopyDlg.h src\app\UI\EditorPage\EmbCanvasEditorPage.h src\app\UI\EditorPage\layers\centeredcheckboxdelegate.h src\app\UI\EditorPage\layers\changelayerdialog.h src\app\UI\EditorPage\layers\layerdata.h src\app\UI\EditorPage\layers\layermodel.h src\app\UI\EditorPage\layers\layerpage.h src\app\UI\EditorPage\layers\layerservice.h src\app\UI\EditorPage\LayerSetDlg.h src\app\UI\EditorPage\ParameterDlg.h src\app\UI\GeneralPage\Location\BasicPositionPage.h src\app\UI\GeneralPage\Location\NeedleDialog.h src\app\UI\GeneralPage\Process\FileList\FileListHelper.h src\app\UI\GeneralPage\Process\FileList\FileListWidget.h src\app\UI\GeneralPage\Process\MainPage.h src\app\UI\MainWindow\MainWindow.h src\app\UI\MainWindow\State\StateWidget.h src\app\UI\MainWindow\Title\TitleWidget.h src\app\UI\ManagePage\Other\OtherPage.h src\app\UI\ManagePage\Software\ExportSelection\ExportSelectionDlg.h src\app\UI\ManagePage\Software\SoftwarePage.h src\app\UI\ManagePage\WorkRecord\WorkRecordPage.h src\app\UI\Shared\ProcessingControlWidget.h src\app\UI\Shared\RunInfo\RunInfoWidget.h src\app\UI\Shared\SharedControl.h src\app\UI\Shared\WorkAssist\WorkAssistWidget.h src\app\UI\SplashScreen\SplashScreen.h src\app\UI\TabWidgetPage\ModuleTabWidget.h src\app\UI\TabWidgetPage\Navigation\LockScreen\LockScreenDlg.h src\app\UI\TabWidgetPage\Navigation\NavigationWidget.h src\app\UI\TabWidgetPage\TabWidgetPages.h src\app\UI\VisionPage\Calibration\CalibrationPage.h src\app\UI\VisionPage\Camera\CameraDoublePage.h src\app\UI\VisionPage\Camera\CameraPage.h src\app\UI\VisionPage\Camera\CameraWidget.h src\app\UI\VisionPage\Share\RegionToolBar.h src\app\UI\VisionPage\Vision\DataEdit\Dialogs\EdgePatrolConfigDlg.h src\app\UI\VisionPage\Vision\DataEdit\Dialogs\ImportShapeDlg.h src\app\UI\VisionPage\Vision\DataEdit\Widgets\EdgePatrolConfigWidget.h src\app\UI\VisionPage\Vision\DataEdit\Widgets\ShapeManagerWidget.h src\app\UI\VisionPage\Vision\DataEdit\Widgets\VisionDataEditWidget.h src\app\UI\VisionPage\Vision\Log\VisionLogWidget.h src\app\UI\VisionPage\Vision\Main\VisionMainControlWidget.h src\app\UI\VisionPage\Vision\Main\VisionMainPage.h src\app\UI\VisionPage\Vision\Main\VisionPage.h src\app\UI\VisionPage\Vision\Options\VisionAdvancedDlg.h src\app\UI\VisionPage\Vision\Options\VisionOptionsDlg.h src\app\Core\UserManager\Widgets\UserLogInDlg.ui src\app\Core\UserManager\Widgets\UserManagerDlg.ui src\app\Core\UserManager\Widgets\UserPWDResetDlg.ui src\app\Module\Alarms\AlarmsDialog.ui src\app\Module\Base\MsgBox.ui src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentDialog.ui src\app\Module\EmbCanvas\EmbCanvasMove.ui src\app\Module\EmbCanvas\EmbCanvasWidget.ui src\app\Module\Register\Widgets\RegisterWidget.ui src\app\UI\DebugPage\Drivers\DriversPages.ui src\app\UI\DebugPage\MaterialTransfer\MaterialTransferPage.ui src\app\UI\DebugPage\RunningInfo\DataSelectDialog.ui src\app\UI\DebugPage\RunningInfo\RunningInfoPage.ui src\app\UI\DebugPage\Workstation\WorkstationPage.ui src\app\UI\EditorPage\CodeColorDlg.ui src\app\UI\EditorPage\CodeInputDlg.ui src\app\UI\EditorPage\CoordinatePointInputDlg.ui src\app\UI\EditorPage\CopyDlg.ui src\app\UI\EditorPage\EmbCanvasEditorPage.ui src\app\UI\EditorPage\layers\changelayerdialog.ui src\app\UI\EditorPage\layers\layerpage.ui src\app\UI\EditorPage\LayerSetDlg.ui src\app\UI\EditorPage\ParameterDlg.ui src\app\UI\GeneralPage\Location\BasicPositionPage.ui src\app\UI\GeneralPage\Location\NeedleDialog.ui src\app\UI\GeneralPage\Process\FileList\FileListWidget.ui src\app\UI\GeneralPage\Process\MainPage.ui src\app\UI\MainWindow\MainWindow.ui src\app\UI\MainWindow\State\StateWidget.ui src\app\UI\MainWindow\Title\TitleWidget.ui src\app\UI\ManagePage\Other\OtherPage.ui src\app\UI\ManagePage\Software\ExportSelection\ExportSelectionDlg.ui src\app\UI\ManagePage\Software\SoftwarePage.ui src\app\UI\ManagePage\WorkRecord\WorkRecordPage.ui src\app\UI\Shared\ProcessingControlWidget.ui src\app\UI\Shared\RunInfo\RunInfoWidget.ui src\app\UI\Shared\WorkAssist\WorkAssistWidget.ui src\app\UI\TabWidgetPage\Navigation\LockScreen\LockScreenDlg.ui src\app\UI\TabWidgetPage\Navigation\NavigationWidget.ui src\app\UI\VisionPage\Calibration\CalibrationPage.ui src\app\UI\VisionPage\Camera\CameraDoublePage.ui src\app\UI\VisionPage\Camera\CameraPage.ui src\app\UI\VisionPage\Camera\CameraWidget.ui src\app\UI\VisionPage\Vision\DataEdit\Dialogs\EdgePatrolConfigDlg.ui src\app\UI\VisionPage\Vision\DataEdit\Dialogs\ImportShapeDlg.ui src\app\UI\VisionPage\Vision\DataEdit\Widgets\EdgePatrolConfigWidget.ui src\app\UI\VisionPage\Vision\DataEdit\Widgets\ShapeManagerWidget.ui src\app\UI\VisionPage\Vision\DataEdit\Widgets\VisionDataEditWidget.ui src\app\UI\VisionPage\Vision\Log\VisionLogWidget.ui src\app\UI\VisionPage\Vision\Main\VisionMainControlWidget.ui src\app\UI\VisionPage\Vision\Main\VisionPage.ui src\app\UI\VisionPage\Vision\Options\VisionAdvancedDlg.ui src\app\UI\VisionPage\Vision\Options\VisionOptionsDlg.ui src\app\Core\AppInfo\AppInfo.cpp src\app\Core\FileDialog\FileDialog.cpp src\app\Core\FileDialog\IPreviewWidget.cpp src\app\Core\FileDialog\PublicFileDialog.cpp src\app\Core\Language\LanguageHelper.cpp src\app\Core\Language\Translator.cpp src\app\Core\Logger\Logger.cpp src\app\Core\SettingsManager\SettingsManager.cpp src\app\Core\SingleApplication\SingleApplication.cpp src\app\Core\Theme\ModernPlatformStyle.cpp src\app\Core\Theme\StyledItemDelegate.cpp src\app\Core\Theme\ThemeManager.cpp src\app\Core\UserManager\UserManager.cpp src\app\Core\UserManager\Widgets\UserLogInDlg.cpp src\app\Core\UserManager\Widgets\UserManagerDlg.cpp src\app\Core\UserManager\Widgets\UserPWDResetDlg.cpp src\app\main.cpp src\app\Module\Alarms\AlarmsDialog.cpp src\app\Module\Alarms\AlarmsManager.cpp src\app\Module\Base\Dialog.cpp src\app\Module\Base\MsgBox.cpp src\app\Module\Base\VariantComboBox.cpp src\app\Module\Clipper2Ex\Clipper2Ex.cpp src\app\Module\Clipper2Ex\Polylabel.cpp src\app\Module\Clipper2Ex\PolyOffseter.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXY.cpp src\app\Module\CommonWidgets\AxisControl\AxisControllerXYEx.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionControlTable.cpp src\app\Module\CommonWidgets\AxisControl\AxisMotionTable.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentButton.cpp src\app\Module\CommonWidgets\AxisControl\CalculateEquivalentDialog.cpp src\app\Module\CommonWidgets\AxisControl\FineTuningStep.cpp src\app\Module\CommonWidgets\AxisControl\ItemDelegate.cpp src\app\Module\CommonWidgets\Ruler\mRuler.cpp src\app\Module\CommonWidgets\Ruler\RulerWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\GaugePanel.cpp src\app\Module\CommonWidgets\SpeedMeter\ObsSpeedMeterWidget.cpp src\app\Module\CommonWidgets\SpeedMeter\SpeedMeterWidget.cpp src\app\Module\CommonWidgets\ViewWidget\TableWidget.cpp src\app\Module\EmbCanvas\ElidedLabel.cpp src\app\Module\EmbCanvas\EmbCanvasController.cpp src\app\Module\EmbCanvas\EmbCanvasMove.cpp src\app\Module\EmbCanvas\EmbCanvasPreviewWidget.cpp src\app\Module\EmbCanvas\EmbCanvasWidget.cpp src\app\Module\EmbCanvas\EmbImportStrategy.cpp src\app\Module\EmbCanvas\GlueCodeComboBox.cpp src\app\Module\Factory\FactoryRegisterHelper.cpp src\app\Module\Graphics\GraphicsCrossItem.cpp src\app\Module\Graphics\GraphicsMeasureItem.cpp src\app\Module\Graphics\GraphicsPointsItem.cpp src\app\Module\Graphics\GraphicsRectResizeHandler.cpp src\app\Module\Graphics\GraphicsScene.cpp src\app\Module\Graphics\GraphicsSelectGridItem.cpp src\app\Module\Graphics\GraphicsSelectRectItem.cpp src\app\Module\Graphics\GraphicsView.cpp src\app\Module\Graphics\OpenGL\GLWidget.cpp src\app\Module\Graphics\OpenGL\ImageGLWidget.cpp src\app\Module\Graphics\OpenGL\Renderer.cpp src\app\Module\Graphics\QGraphicsImageItem.cpp src\app\Module\Graphics\SceneTreeItem.cpp src\app\Module\Graphics\SceneTreeView.cpp src\app\Module\MindVision\CameraGraphicsScene.cpp src\app\Module\MindVision\CameraManager.cpp src\app\Module\MindVision\MindVisionCamera.cpp src\app\Module\MindVision\MindVisionCameraInfo.cpp src\app\Module\MindVision\VideoUtils.cpp src\app\Module\Modbus\DriversAddrListManager.cpp src\app\Module\Modbus\ModbusHelper.cpp src\app\Module\Modbus\SerialPortHelper.cpp src\app\Module\ModuleManager\FunctionCheckBox.cpp src\app\Module\ModuleManager\IFunctionWidget.cpp src\app\Module\ModuleManager\ModuleHelper.cpp src\app\Module\ModuleManager\ModuleManager.cpp src\app\Module\ModuleManager\ObsFunctionCheckBox.cpp src\app\Module\ModuleManager\ObsFunctionRadioButton.cpp src\app\Module\MountWatcher\MountWatcher.cpp src\app\Module\Register\RegInfoOld.cpp src\app\Module\Register\RegisterManager.cpp src\app\Module\Register\SoftRegNew.cpp src\app\Module\Register\SoftRegOld.cpp src\app\Module\Register\Widgets\RegisterWidget.cpp src\app\Module\SoundSetting\SoundService.cpp src\app\Module\Units\ColorHelper.cpp src\app\Module\Units\FileHelper.cpp src\app\Module\Units\GraphicHelper.cpp src\app\Module\Units\ImageHelper.cpp src\app\Module\Units\PathHelper.cpp src\app\Module\Units\ScreenShot.cpp src\app\Module\Units\SqlHelper.cpp src\app\Module\Units\ZipHelper.cpp src\app\Module\Upgrade\TSUpgradeHelper.cpp src\app\Module\VisionSystem\BaseSystem.cpp src\app\Module\VisionSystem\CalibrationSystem.cpp src\app\Module\VisionSystem\Component\ImageProcessor.cpp src\app\Module\VisionSystem\Component\InputComponent.cpp src\app\Module\VisionSystem\Component\LogComponent.cpp src\app\Module\VisionSystem\Component\ObjectLocatorComponent.cpp src\app\Module\VisionSystem\Component\ProcessComponent.cpp src\app\Module\VisionSystem\Component\ShapeArrayToolComponent.cpp src\app\Module\VisionSystem\Component\WorkspaceInput.cpp src\app\Module\VisionSystem\Data\PrimitiveConvertor2.cpp src\app\Module\VisionSystem\Data\ShapeConfig.cpp src\app\Module\VisionSystem\Data\VisionConfig.cpp src\app\Module\VisionSystem\Data\VisionMeta.cpp src\app\Module\VisionSystem\Data\VisionMetaFileCodec.cpp src\app\Module\VisionSystem\Graphics\DoubleGraphicsView.cpp src\app\Module\VisionSystem\Graphics\EdgeItemDelegate.cpp src\app\Module\VisionSystem\Graphics\GraphicsToolBar.cpp src\app\Module\VisionSystem\Graphics\ShapeTableModel.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsScene.cpp src\app\Module\VisionSystem\Graphics\VisionGraphicsView.cpp src\app\Module\VisionSystem\Graphics\VisionSystemGraphics.cpp src\app\Module\VisionSystem\Manager\CalibManager.cpp src\app\Module\VisionSystem\Manager\EdgesTools.cpp src\app\Module\VisionSystem\Manager\ShapeManager.cpp src\app\Module\VisionSystem\Manager\VisionSystemManager.cpp src\app\Module\VisionSystem\Manager\WorkSpaceManager.cpp src\app\Module\VisionSystem\OpenCV\CvCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvCalibTool.cpp src\app\Module\VisionSystem\OpenCV\CvEdgePatrol.cpp src\app\Module\VisionSystem\OpenCV\CvFixedCameraCalibration.cpp src\app\Module\VisionSystem\OpenCV\CvObjectLocator.cpp src\app\Module\VisionSystem\OpenCV\CvShapeArrayTool.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTools.cpp src\app\Module\VisionSystem\OpenCV\OpenCvTypes.cpp src\app\Module\VisionSystem\Process\EdgeProcess.cpp src\app\Module\VisionSystem\Process\WarpProcess.cpp src\app\Module\VisionSystem\Units\AngleDistanceTree.cpp src\app\Module\VisionSystem\Units\VisionGraphicsTools.cpp src\app\Module\VisionSystem\VisionAutoWork.cpp src\app\Module\VisionSystem\VisionSystem.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibration.cpp src\app\Module\VisionSystem\ZSVision\ZsCalibTool.cpp src\app\Module\VisionSystem\ZSVision\ZsEdgePatrol.cpp src\app\Module\VisionSystem\ZSVision\ZsObjectLocator.cpp src\app\Module\VisionSystem\ZSVision\ZsShapeArrayTool.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionManager.cpp src\app\Module\VisionSystem\ZSVision\ZsVisionTypes.cpp src\app\Module\WorkData\WorkDataHelper.cpp src\app\Module\XModem\FileDevice.cpp src\app\Module\XModem\Transport.cpp src\app\Module\XModem\XModem.cpp src\app\Module\XModem\XModemHelper.cpp src\app\UI\DebugPage\AxisMotion\AxisMotion.cpp src\app\UI\DebugPage\Drivers\DriversPages.cpp src\app\UI\DebugPage\Drivers\PulsesObsLabel.cpp src\app\UI\DebugPage\IOControl\IOController.cpp src\app\UI\DebugPage\IOControl\ObsControllerButton.cpp src\app\UI\DebugPage\IOControl\ObsIOPushButton.cpp src\app\UI\DebugPage\MaterialTransfer\MaterialTransferPage.cpp src\app\UI\DebugPage\RunningInfo\DataSelectDialog.cpp src\app\UI\DebugPage\RunningInfo\RunningInfoPage.cpp src\app\UI\DebugPage\Workstation\WorkstationPage.cpp src\app\UI\EditorPage\CodeColorDlg.cpp src\app\UI\EditorPage\CodeInputDlg.cpp src\app\UI\EditorPage\CoordinatePointInputDlg.cpp src\app\UI\EditorPage\CopyDlg.cpp src\app\UI\EditorPage\EmbCanvasEditorPage.cpp src\app\UI\EditorPage\layers\centeredcheckboxdelegate.cpp src\app\UI\EditorPage\layers\changelayerdialog.cpp src\app\UI\EditorPage\layers\layerdata.cpp src\app\UI\EditorPage\layers\layermodel.cpp src\app\UI\EditorPage\layers\layerpage.cpp src\app\UI\EditorPage\layers\layerservice.cpp src\app\UI\EditorPage\LayerSetDlg.cpp src\app\UI\EditorPage\ParameterDlg.cpp src\app\UI\GeneralPage\Location\BasicPositionPage.cpp src\app\UI\GeneralPage\Location\NeedleDialog.cpp src\app\UI\GeneralPage\Process\FileList\FileListHelper.cpp src\app\UI\GeneralPage\Process\FileList\FileListWidget.cpp src\app\UI\GeneralPage\Process\MainPage.cpp src\app\UI\MainWindow\MainWindow.cpp src\app\UI\MainWindow\State\StateWidget.cpp src\app\UI\MainWindow\Title\TitleWidget.cpp src\app\UI\ManagePage\Other\OtherPage.cpp src\app\UI\ManagePage\Software\ExportSelection\ExportSelectionDlg.cpp src\app\UI\ManagePage\Software\SoftwarePage.cpp src\app\UI\ManagePage\WorkRecord\WorkRecordPage.cpp src\app\UI\Shared\ProcessingControlWidget.cpp src\app\UI\Shared\RunInfo\RunInfoWidget.cpp src\app\UI\Shared\SharedControl.cpp src\app\UI\Shared\WorkAssist\WorkAssistWidget.cpp src\app\UI\SplashScreen\SplashScreen.cpp src\app\UI\TabWidgetPage\ModuleTabWidget.cpp src\app\UI\TabWidgetPage\Navigation\LockScreen\LockScreenDlg.cpp src\app\UI\TabWidgetPage\Navigation\NavigationWidget.cpp src\app\UI\TabWidgetPage\TabWidgetPages.cpp src\app\UI\VisionPage\Calibration\CalibrationPage.cpp src\app\UI\VisionPage\Camera\CameraDoublePage.cpp src\app\UI\VisionPage\Camera\CameraPage.cpp src\app\UI\VisionPage\Camera\CameraWidget.cpp src\app\UI\VisionPage\Share\RegionToolBar.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\EdgePatrolConfigDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Dialogs\ImportShapeDlg.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\EdgePatrolConfigWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\ShapeManagerWidget.cpp src\app\UI\VisionPage\Vision\DataEdit\Widgets\VisionDataEditWidget.cpp src\app\UI\VisionPage\Vision\Log\VisionLogWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainControlWidget.cpp src\app\UI\VisionPage\Vision\Main\VisionMainPage.cpp src\app\UI\VisionPage\Vision\Main\VisionPage.cpp src\app\UI\VisionPage\Vision\Options\VisionAdvancedDlg.cpp src\app\UI\VisionPage\Vision\Options\VisionOptionsDlg.cpp src\app\Core\AppInfo\AppInfo.rc -ts src\app\Resource\translations\app_en_US.ts), Unknown Error (206)
1>EXEC : warning : add_requires(opencv 4.11.0): vs_runtime is deprecated, please use runtimes!
1>EXEC : warning : add -v for getting more warnings ..
1>D:\Program Files\xmake\scripts\vsxmake\vsproj\Xmake.targets(97,5): error MSB3073: 命令“...”已退出，代码为 -1。
1>已完成生成项目“GlueHmi.vcxproj”的操作 - 失败。
========== 生成: 0 成功，1 失败，0 最新，0 已跳过 ==========
========== 生成 于 18:39 完成，耗时 09.384 秒 ==========

--- a/xmake/rules/qt/ts/xmake.lua
+++ b/xmake/rules/qt/ts/xmake.lua
@@ -4,22 +4,36 @@ rule("qt.ts")
 
     on_config(function (target)
         import("lib.detect.find_file")
+        import("core.base.json")
 
         -- get source file
         local lupdate_argv = {"-no-obsolete"}
         local sourcefile_ts
+        local source_files = {}
         for _, sourcebatch in pairs(target:sourcebatches()) do
             if sourcebatch.rulename == "qt.ts" then
                 sourcefile_ts = sourcebatch.sourcefiles
             else
                 if sourcebatch.sourcefiles then
                     for _, sourcefile in ipairs(sourcebatch.sourcefiles) do
-                        table.join2(lupdate_argv, path(sourcefile))
+                        table.insert(source_files, sourcefile)
                     end
                 end
             end
         end
         if sourcefile_ts then
+            -- save source files
+            source_files = table.unique(source_files)
+            local json_data = {
+                projectFile = "",
+                sources = source_files
+            }
+
+            local json_path = path.join(target:autogendir(), "rules", "qt", "ts", "sources.json")
+            json.savefile(json_path, json_data)
+
+            table.join2(lupdate_argv, {"-project", path(json_path)})
+
             -- get lupdate and lrelease
             local qt = assert(target:data("qt"), "qt not found!")
 

